tests/unit/ds-macro-test.js

```diff
@@ -1,44 +1,236 @@
+    attributes.push({name: attr, type: meta.type});
+  });
+
+  assert.deepEqual(attributes, [
+    {name: 'electric', type: 'boolean'},
+  ], 'Car attributes are correct');
+
+  let user_attributes = [];
+  User.eachAttribute((attr, meta) => {
+    user_attributes.push({name: attr, type: meta.type});
+  });
+
+  assert.deepEqual(user_attributes, [
+    {name: 'name', type: undefined},
+    {name: 'age',  type: 'number'}
+  ], 'User attributes are correct');
+
+  let user_relationships = [];
+  User.eachRelationship((attr, meta) => {
+    user_relationships.push({key: attr, type: meta.type});
+  });
+
+  assert.deepEqual(user_relationships, [
+    {key: 'cars', type: 'car'}
+  ], 'User relationships are correct');
+});
+
+test('can find records in store with object protos', function(assert) {
+  let Tag = Model.extend({
+    @attr name: null,
+    @hasMany users: null
+  });
+
+  let User = Model.extend({
+    @attr name: null,
+    @belongsTo tag: null
+  });
+
+  let store = createStore({ tag: Tag, user: User });
+
+  run(() => {
+    store.push({
+      data: [{
+        type: 'tag',
+        id: '5',
+        attributes: {
+          name: 'friendly'
+        }
+      }, {
+        type: 'user',
+        id: '1',
+        attributes: {
+          name: 'Tom Dale'
+        },
+        relationships: {
+          tag: {
+            data: { type: 'tag', id: '5' }
+          }
+        }
+      }]
+    });
+  });
+
+  return run(() => {
+    const tag = store.peekRecord('tag', 5);
+    assert.equal(get(tag, 'name'), 'friendly', 'retrieves tag record from store');
+    const user = store.peekRecord('user', 1);
+    assert.equal(get(user, 'name'), 'Tom Dale', 'retrieves user record from store');
+    assert.equal(get(user, 'tag.name'), 'friendly', 'the tag should have name');
+  });
+});
+
+test('can find records in store with ES6 classes', function(assert) {
+  class Tag extends Model {
+    @attr name;
+    @hasMany users;
+  }
+
+  class User extends Model {
+    @attr name;
+    @belongsTo tag;
+  }
+
+  let store = createStore({ tag: Tag, user: User });
+
+  run(() => {
+    store.push({
+      data: [{
+        type: 'tag',
+        id: '5',
+        attributes: {
+          name: 'friendly'
+        }
+      }, {
+        type: 'user',
+        id: '1',
+        attributes: {
+          name: 'Tom Dale'
+        },
+        // relationships: {
+        //   tag: {
+        //     data: { type: 'tag', id: '5' }
+        //   }
+        // }
+      }]
+    });
+  });
+
+  return run(() => {
+    const tag = store.peekRecord('tag', 5);
+    assert.equal(get(tag, 'name'), 'friendly', 'retrieves tag record from store');
+    const user = store.peekRecord('user', 1);
+    assert.equal(get(user, 'name'), 'Tom Dale', 'retrieves user record from store');
+    // assert.equal(get(user, 'tag') instanceof Tag, true, 'the tag property should return a tag');
+    // assert.equal(get(user, 'tag.name'), 'friendly', 'the tag should have name');
```

Comment:
@buschtoens @pzuraq Ah here we go.  So `parentType` is not defined for the relationship model (ED [line](https://github.com/emberjs/data/blob/0999f8a0a756a3118f1e6a4e110709be72e1ad3e/addon/-private/system/model/model.js#L1323)).  Maybe ya'll already knew this :)  See the test above that shows pushing/finding the relationship with object protos work whereas this test doesn't.

Reply:
Sweet, thanks for tracking this down! Will help a lot with figuring out a fix. I should be able to get to this sometime after the holidays